parser.c

```diff
@@ -103,28 +108,39 @@ static int parse_open_string(struct tinyiiod *iiod, char *str)
 		return -EINVAL;
 
 	str = ptr + 1;
+	mask = strtoul(str, NULL, 16);
```

Comment:
the `strtoul` conversion should be it's own commit as well
though, i am not sure we should do it

i suspect the reason for this code [to be as it is] is to make it adaptable for really simple micro-controllers;
`strtoul` is a libc library that may not be present all the time;
maybe, `compat.c / compat.h` file could be added to adapt `strtoul` to systems where this isn't present